/**:
  ros__parameters:
    # -- Node Configurations --
    planning_algorithm: "astar"  # options: astar, rrtstar
    waypoints_velocity: 5.0
    update_rate: 10.0
    th_arrived_distance_m: 0.5
    th_stopped_time_sec: 1.0
    th_stopped_velocity_mps: 0.01
    th_course_out_distance_m: 1.0
    th_obstacle_time_sec: 1.0
    vehicle_shape_margin_m: 0.5
    replan_when_obstacle_found: true
    replan_when_course_out: true

    # -- Configurations common to the all planners --
    # base configs
    time_limit: 30000.0
    max_turning_ratio: 0.5 # ratio of actual turning limit of vehicle
    turning_steps: 1
    # search configs
    theta_size: 120
    angle_goal_range: 6.0
    lateral_goal_range: 0.5
    longitudinal_goal_range: 1.0
    curve_weight: 0.5
    reverse_weight: 0.7
    direction_change_weight: 2.0
    # costmap configs
    obstacle_threshold: 100

    # -- A* search Configurations --
    astar:
      search_method: "forward"  # options: forward, backward
      only_behind_solutions: false
      use_back: true
      adapt_expansion_distance: true
      expansion_distance: 0.5
      near_goal_distance: 3.0
      distance_heuristic_weight: 2.0
      smoothness_weight: 0.5
      obstacle_distance_weight: 1.75
      goal_lat_distance_weight: 5.0

    # -- RRT* search Configurations --
    rrtstar:
      enable_update: true
      use_informed_sampling: true
      max_planning_time: 150.0
      neighbor_radius: 8.0
      margin: 0.1
